pkg/certificates/renewerbottlerocket.go

```diff
@@ -0,0 +1,358 @@
+package certificates
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/aws/eks-anywhere/pkg/logger"
+)
+
+const (
+	brEtcdCertDir           = "/var/lib/etcd"
+	brControlPlaneCertDir   = "/var/lib/kubeadm/pki"
+	brControlPlaneManifests = "/var/lib/kubeadm/manifests"
+	brTempDir               = "/run/host-containerd/io.containerd.runtime.v2.task/default/admin/rootfs/tmp"
+)
+
+// BottlerocketRenewer implements OSRenewer for Bottlerocket systems.
+type BottlerocketRenewer struct {
+	osType OSType
+	backup string
+}
+
+// NewBottlerocketRenewer creates a new BottlerocketRenewer.
+func NewBottlerocketRenewer(backupDir string) *BottlerocketRenewer {
+	return &BottlerocketRenewer{
+		osType: OSTypeBottlerocket,
+		backup: backupDir,
+	}
+}
+
+// RenewControlPlaneCerts renews certificates for control plane nodes.
+func (b *BottlerocketRenewer) RenewControlPlaneCerts(
+	ctx context.Context,
+	node string,
+	cfg *RenewalConfig,
+	component string,
+	ssh SSHRunner,
+) error {
+	logger.V(0).Info("Processing control-plane node", "node", node)
+
+	hasExternalEtcd := cfg != nil && len(cfg.Etcd.Nodes) > 0
+
+	if hasExternalEtcd {
+		if err := b.transferCertsToControlPlane(ctx, node, ssh); err != nil {
+			return fmt.Errorf("transferring certificates to control plane node: %v", err)
+		}
+	}
+
+	sessionCmds := buildBRSheltieCmd(
+		buildBRImagePullCmd(),
+		buildBRControlPlaneBackupCertsCmd(component, hasExternalEtcd, b.backup, brControlPlaneCertDir),
+		buildBRControlPlaneRenewCertsCmd(),
+		buildBRControlPlaneCheckCertsCmd(),
+		buildBRControlPlaneCopyCertsFromTmpCmd(),
+		buildBRControlPlaneRestartPodsCmd(),
+	)
+
+	if _, err := ssh.RunCommand(ctx, node, sessionCmds); err != nil {
+		return fmt.Errorf("renewing control plane certificates: %v", err)
+	}
+
+	logger.V(0).Info("Renewed control-plane certificates", "node", node)
+	return nil
+}
+
+func (b *BottlerocketRenewer) transferCertsToControlPlane(
+	ctx context.Context, node string, ssh SSHRunner,
+) error {
+	logger.V(4).Info("Transferring certificates to control-plane node", "node", node)
+
+	crtB, err := os.ReadFile(filepath.Join(
+		b.backup, tempLocalEtcdCertsDir, "apiserver-etcd-client.crt"))
+	if err != nil {
+		return fmt.Errorf("reading certificate file: %v", err)
+	}
+	keyB, err := os.ReadFile(filepath.Join(
+		b.backup, tempLocalEtcdCertsDir, "apiserver-etcd-client.key"))
+	if err != nil {
+		return fmt.Errorf("reading key file: %v", err)
+	}
+
+	sessionCmds := buildBRSheltieCmd(
+		buildBRCreateTmpDirCmd(tempLocalEtcdCertsDir),
```

Comment:
instead of creating and writing one file line by line, can we copy the entire directory to /tmp?